File: src/main/java/tk/skulk/plugin/core/extensions/entityoverride/listeners/EndermanNoGriefListener.java
package tk.skulk.plugin.core.extensions.entityoverride.listeners;

import org.bukkit.entity.EntityType;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.entity.EntityChangeBlockEvent;
import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.extensions.entityoverride.EntityOverrideExtension;
import tk.skulk.plugin.wrapper.BaseListener;

public final class EndermanNoGriefListener extends BaseListener<EntityOverrideExtension> {
    public EndermanNoGriefListener(final @NotNull EntityOverrideExtension extension) {
        super(extension);
    }

    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    public void onEntityChangeBlockEvent(final @NotNull EntityChangeBlockEvent event) {
        if (event.getEntityType() != EntityType.ENDERMAN) {
            return;
        }

        event.setCancelled(true);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/entityOverride/listeners/EndermanNoGriefListener.kt
package tk.skulk.plugin.core.extensions.entityOverride.listeners

import org.bukkit.entity.EntityType
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.entity.EntityChangeBlockEvent
import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.entityOverride.EntityOverrideExtension
import tk.skulk.plugin.framework.Listener

class EndermanNoGriefListener(extension: EntityOverrideExtension) : Listener<SNPlugin, EntityOverrideExtension>(
    extension
) {
    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    fun onEntityChangeBlock(event: EntityChangeBlockEvent) {
        if (event.entityType != EntityType.ENDERMAN) {
            return
        }

        event.isCancelled = true
    }
}
